### ISSUE TITLE
[Workflow notifications] Notifications not processed after clean install

### ISSUE THREAD

**Issue by graham-web** [localgovdrupal/localgov_workflows#92]
Testing a clean install of the `localgov_workflows_notifications` module, we noticed that the hook_cron logic to enqueue entities won't ever run, due to the state variable `localgov_workflows_notifications.last_email_run` not having been initialised.

In the below code, the default value (current request time) will always be before the calculated "next run" time, making the if-condition fail:
```
 $request_time = \Drupal::time()->getRequestTime();
  $last_run = \Drupal::state()->get('localgov_workflows_notifications.last_email_run', $request_time);
  $next_run = $last_run + 86400 * $settings->get('email_frequency');
  if ($request_time >= $next_run) {
```

### Resolutions

#90 suggests that the state would be initialised to the install time of the module - should it do this?

Else, amending the if condition to check for a default value would fix the issue. e.g.:

`  if ($request_time >= $next_run || $request_time == $last_run) { `

**Comment by stephen-cox** [comment]
Hi @graham-web, thanks for reporting this rather serious bug.

I had initially just set the `$last_run` to 0 if it wasn't set, but this triggers alerts for all content that is currently in needs review.

Initializing `last_email_run` when the module is enabled to the current time makes sense for now, although this might change with #90. I'll also add an update hook to set it for current installs.

**Comment by stephen-cox** [comment]
This should be fixed with #93
